hs_core/tests/api/views/test_authorize.py

```diff
@@ -0,0 +1,856 @@
+                      # or published
+                      {'res_id': self.res.short_id, 'needed_permission': ACTION_TO_AUTHORIZE.EDIT_RESOURCE,
+                       'success': True, 'exception': None},
+                      # super user has no authorization for creating a resource version
+                      {'res_id': self.res.short_id, 'needed_permission': ACTION_TO_AUTHORIZE.CREATE_RESOURCE_VERSION,
+                       'success': False, 'exception': PermissionDenied}
+                      ]
+
+        parameters = [
+                      # super user has authorization for setting resource flags (e.g., public,
+                      # published/immutable, shareable etc) for a private, public or discoverable resource
+                      {'res_id': self.res.short_id, 'needed_permission': ACTION_TO_AUTHORIZE.SET_RESOURCE_FLAG,
+                       'success': True, 'exception': None},
+                      # super user has authorization for deleting a private, public or discoverable resource
+                      {'res_id': self.res.short_id, 'needed_permission': ACTION_TO_AUTHORIZE.DELETE_RESOURCE,
+                       'success': True, 'exception': None},
+                      ] + common_parameters
+
+        # test for private resource
+        self.assertFalse(self.res.raccess.public)
+        self.assertFalse(self.res.raccess.discoverable)
+
+        self._run_tests(self.request, parameters)
+
+        # test for discoverable resource
+        self.assertFalse(self.res.raccess.discoverable)
+        self.res.raccess.discoverable = True
+        self.res.raccess.save()
+
+        self._run_tests(self.request, parameters)
+
+        # test for public resource
+        self.assertTrue(self.res.raccess.discoverable)
+        self.assertFalse(self.res.raccess.public)
+        self.res.raccess.discoverable = False
+        self.res.raccess.public = True
+        self.res.raccess.save()
+
+        self._run_tests(self.request, parameters)
+
+        # test for immutable/published resource
+        self.assertFalse(self.res.raccess.immutable)
+        self.assertFalse(self.res.raccess.published)
+        self.assertTrue(self.res.raccess.public)
+
+        self.res.raccess.immutable = True
+        self.res.raccess.published = True
+        self.res.raccess.save()
+
+        parameters = [
+                      # super user has authorization for setting resource flags (e.g., public,
+                      # published/immutable, shareable etc) for a published resource
+                      {'res_id': self.res.short_id, 'needed_permission': ACTION_TO_AUTHORIZE.SET_RESOURCE_FLAG,
+                       'success': True, 'exception': None},
+                      # super user has no authorization for deleting a published resource
+                      {'res_id': self.res.short_id, 'needed_permission': ACTION_TO_AUTHORIZE.DELETE_RESOURCE,
+                       'success': False, 'exception': PermissionDenied}
```

Comment:
@pkdash This needs to be changed in the parameters since superuser can delete a published resource. A PermissionDefined should not be raised and success should be true. I think this DELETE_RESOURCE action for super user can be moved to common parameters above for testing.